nixos/modules/services/x11/display-managers/gdm.nix

```diff
@@ -335,8 +358,10 @@ in
         auth       requisite                   pam_faillock.so      preauth
         auth       required                    ${pkgs.fprintd}/lib/security/pam_fprintd.so
         auth       required                    pam_env.so
+        ${lib.optionalString config.boot.initrd.systemd.enable ''
+          auth       optional ${config.systemd.package}/lib/security/pam_systemd_loadkey.so${loadkeyArgs}
+        ''}
         ${lib.optionalString pamCfg.login.enableGnomeKeyring ''
-          auth       [success=ok default=1]      ${pkgs.gnome.gdm}/lib/security/pam_gdm.so
```

Comment:
Wouldn't pam_gdm cover the case where systemd is not enabled in initrd?

Reply:
I don't think the kernel keyring is populated in that case (unless the initrd uses systemd-ask-password)

Reply:
I believe the code for that is here (could be mistaken though): https://github.com/NixOS/nixpkgs/blob/3563397b2f10ffa1891e1a6ce99d13d960d73acd/nixos/modules/system/boot/luksroot.nix#L177

Reply:
so as far as I know, neither `pam_gdm` nor `pam_systemd_loadkey` will work without systemd in the initrd

Reply:
I don't mind leaving out the `config.boot.initrd.systemd.enable` check and just putting everything inside `pamCfg.login.enableGnomeKeyring` though